### DISCUSSION TITLE
Flow pallet

### DISCUSSION THREAD

**Discussion by vovacha** [gamedaoco/gamedao-protocol#84] [General]
Let's discuss anything related to the architecture/development of the Flow pallet (fundraising).

**Comment by vovacha** [comment]
What is the purpose of the `update_state` extrinsic? Any manual state change will result in unpredictable campaign behavior (not finalized or not activated, etc.).

These are the main campaign's states:

- Created
- Activated
- Paused
- Succeeded
- Failed
- Locked

**Created** -> **Activated** works on reaching the start block.
**Activated** -> **Succeeded** (**Failed**) works on reaching the expiry block.

Only **Pause** could actually be the case for `update_state`. I assume it means stop gathering contributions?
What is the **Locked** status about?

@5-mark:
> So far I know that Paused and Locked should be the statuses GameDAO DAO shall be able to set.
If sth. is wrong with a campaign or their initiators GameDAO puts a veto and can "lock" the campaign. I guess its not implemented yet that when having theses statuses contributions are not possible.

@2075:
> update state was/is needed to execute external action e.g. through a council vote and related.

**Reply by vovacha** [reply]
We are not using FRAME abstractions like councils. If our internal governance was meant, it shouldn't be extrinsic but a public function, exposable through the trait.
And mainly, the question was about the states - which transitions are possible? Due to its current implementation, if a campaign is not expired and not finalized yet, the status can be changed to anything, such as Succeded or Failed, resulting in unpredicted behavior - contributions processing, etc.

**Comment by 2075** [comment]
paused == under discussion, pending council decision, can be unpaused or failed and therefore reverted.
locked == authority lock due to e.g. legal actions, community votes. similar to a circuit breaker.

**Reply by vovacha** [reply]
I suggest removing the `update_state` extrinsic for the following reasons:
- Paused / Locked statuses processing logic doesn't exist yet. Using this extrinsic could only lead to unpredicted outcomes.
- Pausing or locking, I assume, will be triggered not directly with extrinsic but through governance mechanisms.
